nilearn/tests/test_signal.py

```diff
@@ -526,52 +563,81 @@ def test_clean_kwargs():
             "butterworth__padlen": 20,
             "butterworth__order": 10,
         },
-    ]
+    ],
+)
+def test_clean_kwargs(kwarg_set):
+    """Providing kwargs to clean should change the filtered results."""
+    n_samples = 34
+    n_features = 501
+    x_orig = generate_signals_plus_trends(
+        n_features=n_features, n_samples=n_samples
+    )
+
     # Base result
     t_r, high_pass, low_pass = 0.8, 0.01, 0.08
     base_filtered = clean(
         x_orig, t_r=t_r, low_pass=low_pass, high_pass=high_pass
     )
-    for kwarg_set in kwargs:
-        test_filtered = clean(
-            x_orig,
-            t_r=t_r,
-            low_pass=low_pass,
-            high_pass=high_pass,
-            **kwarg_set,
-        )
-        # Check that results are **not** the same.
-        assert np.any(np.not_equal(base_filtered, test_filtered))
+
+    test_filtered = clean(
+        x_orig,
+        t_r=t_r,
+        low_pass=low_pass,
+        high_pass=high_pass,
+        **kwarg_set,
+    )
+
+    # Check that results are **not** the same.
+    assert np.any(np.not_equal(base_filtered, test_filtered))
 
 
 def test_clean_frequencies():
-    """Using butterworth method."""
+    """Check several values for low and high pass."""
+    sx1 = np.sin(np.linspace(0, 100, 2000))
+    sx2 = np.sin(np.linspace(0, 100, 2000))
+    sx = np.vstack((sx1, sx2)).T
+
+    t_r = 2.5
+    standardize = False
+
+    cleaned_signal = clean(
+        sx, standardize=standardize, high_pass=0.002, low_pass=None, t_r=t_r
+    )
+    assert cleaned_signal.max() > 0.1
+
+    cleaned_signal = clean(
+        sx, standardize=standardize, high_pass=0.2, low_pass=None, t_r=t_r
+    )
+    assert cleaned_signal.max() < 0.01
+
+    cleaned_signal = clean(sx, standardize=standardize, low_pass=0.01, t_r=t_r)
+    assert cleaned_signal.max() > 0.9
```

Comment:
Optionally this part could be parametrized  with values for `high_pass`, `low_pass` and expected_value for `cleaned_signal.max()`.

Reply:
Yeah I thought about it but sometimes there is > or a < the assert part so i decided to not refactor more than this. For now...